### PR TITLE
Add release job for runtime-watcher

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Release job for the watcher.
It is created similar to the existing PR building job: `pull-runtime-watcher-img-build` and existing release jobs for other components - for example: `rel-istio-build`.
Simplification suggested by @dekiel (removing Signify dev secret) is applied.

Changes proposed in this pull request:

- Add release job for runtime-watcher

**Related issue(s)**
Fixes: [#182 ](https://github.com/kyma-project/runtime-watcher/issues/182)

### CODE REVIEW

**Comment by Tomasz-Smelcerz-SAP** [comment]
Note for the reviewer: pjtester doesn't work for this one, probably because of security containers are not mounted for pjtester jobs. We can just merge this one as it doesn't interfere with anything - and test it manually.

File: prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml [hunk 1]
```diff
@@ -96,3 +96,61 @@ postsubmits:
...
+                mountPath: /secret
+                readOnly: true
+        volumes:
+          - name: share
+          - name: config
+            configMap:
+              name: kaniko-build-config
+          - name: signify-secret
+            secret:
+              secretName: signify-dev-secret
```
Comment (dekiel) [inline prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml:156]: Signify dev secret is not need.
```suggestion
            volumeMounts:
              - name: share
                mountPath: /home/user/.local/share/buildkit
              - name: config
                mountPath: /config
                readOnly: true
        volumes:
          - name: share
          - name: config
            configMap:
              name: kaniko-build-config
```

File: prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml [hunk 2]
```diff
@@ -96,3 +96,61 @@ postsubmits:
...
+      always_run: true
+      skip_report: false
+      decorate: true
+      cluster: trusted-workload
+      max_concurrency: 10
+      branches:
+        - ^\d+\.\d+\.\d+(?:-.*)?$
+      spec:
+        containers:
+          - image: "eu.gcr.io/sap-kyma-neighbors-dev/image-builder:v20230313-8dfce5f0b-buildkit"
```
Comment (dekiel) [inline prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml:118]: For what you need a buildkit builder? Can you use kaniko builder instead?

**Review by dekiel — COMMENTED** [review]
Neighbours review is not needed here.

**Review by dekiel — DISMISSED** [review]

File: prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml:118]: @dekiel What is the difference between the two?

File: prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml (hunk 1 above)
Comment (Tomasz-Smelcerz-SAP) [reply prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml:156]: > Signify dev secret is not need.

I copied this from other teams' configs (e.g. Istio). If they use it for release builds, why it's not needed here?

Comment (dekiel) [reply prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml:156]: It's not needed in configs of other teams too. This was used in the past as we were signing images with dev instance of signify. At present this is not true anymore and signify dev secret is not used. This is some configuration which was not cleaned when change was applied.

File: prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml (hunk 2 above)
Comment (dekiel) [reply prow/jobs/kyma-project/runtime-watcher/runtime-watcher.yaml:118]: Buildkit builder was added to make possible building of images for different architectures like arm64. Later the buildkit image was enhanced with bash and used for some very specific build process. Both of this use cases are deprecated and won't be supported in ADO backend. I'm asking to know if this is some special use case that you need a buildkit version.

**Pull request merged**